ship less files and simplify gemspec

grosser opened the pull request:
@jnicklas 

shipping a bunch of test / misc files means download takes longer and vendor/cache gets bigger

adding spec to require-path means the spec files are actually in the gems load path ... but they are not supposed to be required ... right ?

also removing LOAD_PATH modifications

jnicklas commented:
The test files are actually included in the gem very deliberately. It's so that backend implementations can require the spec files and run Refile's test suite against their own implementation. See e.g. https://github.com/refile/refile-s3/blob/master/spec/refile/s3_spec.rb#L1

The rest of the Gemfile is just want bundler autogenerates.

I think it's certainly possible to slim down the gem a little bit (we don't need the bundled Rails app to run feature tests for example), but is this really that much of an issue?

grosser commented:
I like my gems as slim as possible ... I'll take another look ...

On Thu, Jun 4, 2015 at 1:55 AM, Jonas Nicklas <notifications@github.com>
wrote:

> The test files are actually included in the gem very deliberately. It's so
> that backend implementations can require the spec files and run Refile's
> test suite against their own implementation. See e.g.
> https://github.com/refile/refile-s3/blob/master/spec/refile/s3_spec.rb#L1
>
> The rest of the Gemfile is just want bundler autogenerates.
>
> I think it's certainly possible to slim down the gem a little bit (we
> don't need the bundled Rails app to run feature tests for example), but is
> this really that much of an issue?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/refile/refile/pull/266#issuecomment-108792804>.
>

grosser commented:
updated, looks good ?

a good chunk of saving when just excluding test_app
`git ls-files lib spec Readme.md`.split($/).reject { |f|
f.include?("test_app") }.size
=> 38
>> `git ls-files lib spec Readme.md`.split($/).size
=> 62

and half of master:
git ls-files | wc -l
      79

On Thu, Jun 4, 2015 at 8:25 AM, Michael Grosser <michael@grosser.it> wrote:

> I like my gems as slim as possible ... I'll take another look ...
>
> On Thu, Jun 4, 2015 at 1:55 AM, Jonas Nicklas <notifications@github.com>
> wrote:
>
>> The test files are actually included in the gem very deliberately. It's
>> so that backend implementations can require the spec files and run Refile's
>> test suite against their own implementation. See e.g.
>> https://github.com/refile/refile-s3/blob/master/spec/refile/s3_spec.rb#L1
>>
>> The rest of the Gemfile is just want bundler autogenerates.
>>
>> I think it's certainly possible to slim down the gem a little bit (we
>> don't need the bundled Rails app to run feature tests for example), but is
>> this really that much of an issue?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/refile/refile/pull/266#issuecomment-108792804>.
>>
>
>

jnicklas commented:
Currently fails due to annoying rubucop errors. Sorry.

grosser commented:
Is ignoring english variables ok ? :)

grosser commented:
@jnicklas good to go ?

grosser commented:
What do you think about moving the reused spec files into lib instead ?

jnicklas commented:
@grosser I chose not to do that, because of my bad experience with Capybara in that regard. Can't count  the number of times people didn't find the spec files because they were in `lib`. :(
